tests/tests/debug_api.rs

```diff
@@ -44,116 +44,40 @@ async fn test_raw_transaction(#[future] katana: Katana, _setup: ()) {
-    assert_eq!(transaction, rpc_transaction);
 
-    // Legacy
-    let reqwest_client = reqwest::Client::new();
     let res = reqwest_client
         .post(format!("http://localhost:{}", server_addr.port()))
         .header("Content-Type", "application/json")
-        .body(
-            RawRpcParamsBuilder::new("debug_getRawTransaction")
-                .add_param(format!("0x{:064x}", *LEGACY_TX_HASH))
-                .build(),
-        )
-        .send()
-        .await
-        .expect("Failed to call Debug RPC");
-    let response = res.text().await.expect("Failed to get response body");
-    let raw: Value = serde_json::from_str(&response).expect("Failed to deserialize response body");
-    let rlp_bytes: Option<Bytes> = serde_json::from_value(raw["result"].clone()).expect("Failed to deserialize result");
-    assert!(rlp_bytes.is_some());
-    // We can decode the RLP bytes to get the transaction and compare it with the original transaction
-    let transaction = TransactionSigned::decode_enveloped(&mut rlp_bytes.unwrap().as_ref()).unwrap();
-    let signer = transaction.recover_signer().unwrap();
-    let transaction = from_recovered_with_block_context(
-        TransactionSignedEcRecovered::from_signed_transaction(transaction, signer),
-        *BLOCK_HASH,
-        BLOCK_NUMBER,
-        None,
-        0,
-    );
-    let res = reqwest_client
-        .post(format!("http://localhost:{}", server_addr.port()))
-        .header("Content-Type", "application/json")
-        .body(
-            RawRpcParamsBuilder::new("eth_getTransactionByHash")
-                .add_param(format!("0x{:064x}", *LEGACY_TX_HASH))
-                .build(),
-        )
+        .body(RawRpcParamsBuilder::new("eth_getTransactionByHash").add_param(format!("0x{:064x}", tx.hash)).build())
         .send()
         .await
         .expect("Failed to call Debug RPC");
     let response = res.text().await.expect("Failed to get response body");
     let response: Value = serde_json::from_str(&response).expect("Failed to deserialize response body");
     let rpc_transaction: reth_rpc_types::Transaction =
         serde_json::from_value(response["result"].clone()).expect("Failed to deserialize result");
+
+    // As per https://github.com/paradigmxyz/reth/blob/603e39ab74509e0863fc023461a4c760fb2126d1/crates/rpc/rpc-types-compat/src/transaction/signature.rs#L17
+    if rpc_transaction.transaction_type.unwrap() == 0 {
+        transaction.signature = Some(reth_rpc_types::Signature {
+            y_parity: rpc_transaction.signature.unwrap().y_parity,
+            ..transaction.signature.unwrap()
+        });
+    }
+
+    // For EIP1559, `gas_price` is recomputed as per https://github.com/paradigmxyz/reth/blob/603e39ab74509e0863fc023461a4c760fb2126d1/crates/rpc/rpc-types-compat/src/transaction/mod.rs#L54
+    if rpc_transaction.transaction_type.unwrap() == 2 {
+        transaction.gas_price = rpc_transaction.gas_price;
+    }
```

Comment:
Why do we need this? Does this perhaps indicate that we are doing something wrong elsewhere?

Reply:
Honestly I don't understand these calculations in reth and I don't feel like they apply to us. In fact it is in the `from_recovered_with_block_context` function that these strange calculations are performed:
- For the parity, in the function here https://github.com/paradigmxyz/reth/blob/603e39ab74509e0863fc023461a4c760fb2126d1/crates/rpc/rpc-types-compat/src/transaction/signature.rs#L17, the parity is set to `None` for Legacy transactions. But on the one hand, in the arbitrary definition of transactions, if I set to `None` for Legacy transactions then the signature test is no longer valid. On the other hand I don't understand why the parity would be `None` because it does exist and is also used to calculate the `v` using the chain ID too...
- For the `gas_price`, I have the impression that they also perform a recalculation of the gas price here https://github.com/paradigmxyz/reth/blob/603e39ab74509e0863fc023461a4c760fb2126d1/crates/rpc/rpc-types-compat/src/transaction/mod.rs#L54 with the tip and the base fee of the block but I don't have the impression that it is applicable for us right?

So in conclusion I just filled these two fields with the solutions to pass the test because I don't have the impression that it is something that matters to us. What do you think?

Reply:
I was actually looking at the code for the parity and also feel like the parity should only be set for transactions that aren't legacy, since for legacy transactions, you can recover the parity by doing `(v - 35)%2` which means you don't need the parity right? What I mean by this is that for legacy transactions, the parity is encoded in the `v` field.

For the gas price I am not too sure, we can leave it like this I guess then.

Reply:
Yes I agree this is in the `v` field but here in fact the problem is just that the `transaction_by_hash` function will only search in the database without doing any calculation. However, in our database, we generate arbitrary transactions using the `arbitrary_with_optional_fields` function. And in this function, for Legacy transactions, we set the parity field.

I thought it was maybe useless and I tried to set it to `None` in this arbitrary transaction generation function like:

```rust
signature: Some(reth_rpc_types::Signature {
                r: transaction_signed.signature.r,
                s: transaction_signed.signature.s,
                v: if transaction_signed.is_legacy() {
                    U256::from(transaction_signed.signature.v(transaction_signed.chain_id()))
                } else {
                    U256::from(transaction_signed.signature.odd_y_parity)
                },
                y_parity: if transaction_signed.is_legacy() {None} else {Some(transaction_signed.signature.odd_y_parity.into()},
            }),
```



but in this case the `assert_eq!(transaction.from, transaction_signed.recover_signer().unwrap());` test no longer passes...

What we could do is at the end of the `transaction_by_hash` function, do an if Legacy, then set parity to None, but it's a bit overkilled right? I think we can say that field parity is "not annoying" right?